### PR TITLE
chore(deps): bump k8s libs to v0.26

### LINKED ISSUE #3250: Bump k8s libraries to 1.26/0.26
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem Statement

Bump the k8s libraries in KIC to v0.26.

The latest *released* version of controller-runtime isn't yet compatible with 0.26 libraries so it might be necessary to evaluate use of an unreleased commit from main (to be corrected in a follow-up patch release of KIC)

### Proposed Solution

_No response_

### Additional information

_No response_

### Acceptance Criteria

- [ ] KIC uses v0.26 Kubernetes libraries

### PR DESCRIPTION
**Pull request by czeslavo**
**What this PR does / why we need it**:

- bumps k8s libs to v0.26.0 
- adds type mapping for backward-compatible ingress addresses updates 
- replaces `controller-runtime` with the latest main ref that already supports v0.26 k8s libs
- aligns meshdetect tests due to `controller-runtime`'s fake client changes (it now requires an index to be existent to `List` objects using a field selector)

**Which issue this PR fixes**:

Fixes #3250.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,7 +2,13 @@ module github.com/kong/kubernetes-ingress-controller/v2
 
 go 1.19
 
-replace github.com/imdario/mergo v0.3.12 => github.com/Kong/mergo v0.3.13
+replace (
+	github.com/imdario/mergo v0.3.12 => github.com/Kong/mergo v0.3.13
+
+	// replacing controller-runtime with a version compatible with v0.26.0 k8s libs
+	// TODO: remove when controller-runtime gets released
```
[the remark this answers is not in the document]
Comment (czeslavo) [reply go.mod:9]: Yeah, sure: c43309b

File: internal/controllers/utils/conversion.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+	case *kongv1beta1.TCPIngress:
+		obj.Status.LoadBalancer.Ingress = netV1ToCoreV1LoadBalancerIngress(newAddresses)
+	case *kongv1beta1.UDPIngress:
+		obj.Status.LoadBalancer.Ingress = netV1ToCoreV1LoadBalancerIngress(newAddresses)
+	case *netv1beta1.Ingress:
+		obj.Status.LoadBalancer.Ingress = netV1ToV1beta1LoadBalancerIngress(newAddresses)
+	case *extv1beta1.Ingress:
+		obj.Status.LoadBalancer.Ingress = netV1ToExtV1beta1LoadBalancerIngress(newAddresses)
+	default:
+		return false, errors.Errorf("unsupported ingress type: %T", obj)
```
Comment (pmalek) [inline internal/controllers/utils/conversion.go:46]: Can we use `errors` or if you want to format the error with error string then `fmt.Errorf`? `github.com/pkg/errors` is already in read-only mode and is not meant to be supported in the long run

Comment (czeslavo) [reply internal/controllers/utils/conversion.go]: Good catch, let's prevent it in the future: 5d19561

File: internal/controllers/utils/conversion_test.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+				copiedOld := old.DeepCopyObject()
+				updatedNeeded, err := UpdateLoadBalancerIngress(old, newAddresses)
+				require.NoError(t, err)
+				assert.True(t, updatedNeeded)
+				assert.NotEqual(t, copiedOld, old, "when updated needed, the old object should be updated")
+			})
+		}
+	})
+
+	t.Run("unknown type passed should not panic", func(t *testing.T) {
```
Comment (pmalek) [inline internal/controllers/utils/conversion_test.go:132]: Nitpick: this subtest name suggest that we're actually testing not panicking which IMHO should be done by something like [`assert.NotPanics()`](https://pkg.go.dev/github.com/stretchr/testify@v1.8.1/assert#NotPanics)

**Review by pmalek — APPROVED** [review]

File: internal/controllers/utils/conversion_test.go (hunk 3 above)
Comment (czeslavo) [reply internal/controllers/utils/conversion_test.go:132]: Let's add it for clarity then: [9d04f85](https://github.com/Kong/kubernetes-ingress-controller/pull/3251/commits/9d04f85b32b723f46c3ed50701556421a8593300)

File: internal/controllers/utils/conversion_test.go [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+		newAddresses := []netv1.IngressLoadBalancerIngress{
+			{
+				IP:       oldIP,
+				Hostname: oldHostname,
+			},
+		}
+		assert.NotPanics(t, func() {
+			_, err := UpdateLoadBalancerIngress(unknownObject, newAddresses)
+			require.ErrorContains(t, err, "unsupported ingress type")
+		})
```
Comment (pmalek) [inline internal/controllers/utils/conversion_test.go:143]:
```suggestion
		var err error
		assert.NotPanics(t, func() {
			_, err = UpdateLoadBalancerIngress(unknownObject, newAddresses)
		})
		require.ErrorContains(t, err, "unsupported ingress type")
```

**Review by pmalek — APPROVED** [review]

**Pull request merged**